Sources/Quick/QuickSpec.m

```diff
@@ -115,10 +115,23 @@ + (SEL)addInstanceMethodForExample:(Example *)example {
     }
 
     const char *types = [[NSString stringWithFormat:@"%s%s%s", @encode(id), @encode(id), @encode(SEL)] UTF8String];
-    NSString *selectorName = [NSString stringWithFormat:@"%@_%@_%ld",
-                              example.name.qck_selectorName,
-                              sanitizedFileName,
-                              (long)example.callsite.line];
+    
+    NSString *originalName = example.name.qck_selectorName;
+    NSString *selectorName = originalName;
+    unsigned int i = 2;
+    
+    static NSMutableSet<NSString *> *selectorNames;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        selectorNames = [[NSMutableSet alloc] init];
+    });
+    
+    while ([selectorNames containsObject:selectorName]) {
+        selectorName = [NSString stringWithFormat:@"%@_%u", originalName, i++];
+    }
```

Comment:
This should probably check whether a number needs to be appended to the end of the selector at all. Check this out:

<img width="537" alt="screen shot 2016-01-23 at 9 59 03 pm" src="https://cloud.githubusercontent.com/assets/552921/12534226/b4d62796-c21c-11e5-80bf-2fb5b7fe336b.png">

Ideally if there are no duplicate names, then the selector stays the same. So in the above example, the selector name should remain `is_a_test`.